### PR TITLE
[FSTORE-647][FSTORE-646][FSTORE-644] Bigquery and GCS python validation and path fix

### PR DESCRIPTION
**Pull request by DhananjayMukhedkar**
This PR adds/fixes/changes...
- please summarize your changes to the code 
- and make sure to include all changes to user-facing APIs

JIRA Issue: -

Priority for Review: -

Related PRs: -

**How Has This Been Tested?**

- [x] Unit Tests
- [ ] Integration Tests
- [x] Manual Tests on VM

**Checklist For The Assigned Reviewer:**

```
- [ ] Checked if merge conflicts with master exist
- [ ] Checked if stylechecks for Java and Python pass
- [ ] Checked if all docstrings were added and/or updated appropriately
- [ ] Ran spellcheck on docstring
- [ ] Checked if guides & concepts need to be updated
- [ ] Checked if naming conventions for parameters and variables were followed
- [ ] Checked if private methods are properly declared and used
- [ ] Checked if hard-to-understand areas of code are commented
- [ ] Checked if tests are effective
- [ ] Built and deployed changes on dev VM and tested manually
- [x] (Checked if all type annotations were added and/or updated appropriately)
```

### CODE REVIEW

File: python/hsfs/storage_connector.py [hunk 1]
```diff
@@ -1244,14 +1248,29 @@ def read(
         # Returns
             `Dataframe`: A Spark dataframe.
         """
-
+        if not engine.get_instance().is_connector_type_supported(
+            StorageConnector.BIGQUERY
+        ):
```
Comment (bubriks) [inline python/hsfs/storage_connector.py:1253]: Maybe this is cleaner
```suggestion
        if not engine.get_instance().is_connector_type_supported(self.type):
```

File: python/hsfs/storage_connector.py [hunk 2]
```diff
@@ -1089,6 +1089,10 @@ def read(
         # Returns
             `Dataframe`: A Spark dataframe.
         """
+        if not engine.get_instance().is_connector_type_supported(StorageConnector.GCS):
```
Comment (bubriks) [inline python/hsfs/storage_connector.py:1092]: Maybe this is cleaner
```suggestion
        if not engine.get_instance().is_connector_type_supported(self.type):
```

**Review by bubriks — COMMENTED** [review]
Just some minor comments

**Comment by DhananjayMukhedkar** [comment]
Cover three JIRAs. They are not mutually related

- https://hopsworks.atlassian.net/browse/FSTORE-647
- https://hopsworks.atlassian.net/browse/FSTORE-644
- https://hopsworks.atlassian.net/browse/FSTORE-646

**Review by bubriks — CHANGES_REQUESTED** [review]
some minor comments, but only duplication is an issue

**Pull request merged**